### PR TITLE
Wallet activity

### PR DESCRIPTION
**Pull request by Jibz1**
Activities section 

https://www.figma.com/file/OzLaRFzevjxdQAbybWEZk0/Sui-Visualize?node-id=4438%3A103142

https://user-images.githubusercontent.com/8676844/184028124-15c4029f-e680-4fea-880b-82a5d13d0e28.mov

### CODE REVIEW

**Review by mystie711 — COMMENTED** [review]
👍🏽 Looking good for the purpose of this task @Jibz1!

I am assuming where I see the delta from Figma...it is just the data not being available right now? For ex. the NFT price inside the summary card at the top.

This prototype makes it pretty clear that we need to add some animated transition sooner than later when switching context from the activity item in list view to the activity receipt detail page. Otherwise it feels a bit disorienting. Will add it to design to-do.

**Comment by Jibz1** [comment]
> 👍🏽 Looking good for the purpose of this task @Jibz1!
> 
> I am assuming where I see the delta from Figma...it is just the data not being available right now? For ex. the NFT price inside the summary card at the top.
> 
> This prototype makes it pretty clear that we need to add some animated transition sooner than later when switching context from the activity item in list view to the activity receipt detail page. Otherwise it feels a bit disorienting. Will add it to design to-do.

Yes, I did not add features where we do not have the data. I will look into the animations next week.

File: wallet/src/ui/app/hooks/useFormatDate.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO - handle multiple date formats
+// Wed Aug 05
+export default function useFormatDate(date: Date) {
+    return new Date(date).toLocaleDateString('en-us', {
```
Comment (Jordan-Mysten) [inline wallet/src/ui/app/hooks/useFormatDate.ts:7]: If you pass `undefined` as the first argument it should use the user's preferred locale, which might be nice.

Also, you don't need to date wrap date, as it's already a date. Should just be able to do `date.toLocaleDateString()`

File: wallet/src/ui/app/pages/home/receipt/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import st from './ReceiptPage.module.scss';
+
+// Response pages for all transactions
+// use txDigest for the transaction result
+function ReceiptPage() {
+    const [searchParams] = useSearchParams();
+
+    // get tx results from url params
+    const txDigest = useMemo(
```
Comment (Jordan-Mysten) [inline wallet/src/ui/app/pages/home/receipt/index.tsx:26]: These memos over search params are probably equally as costly as the underlying `get` call itself, so you really don't need them.

File: wallet/src/ui/app/pages/home/nft-details/transfer-nft/index.tsx [hunk 3]
```diff
@@ -111,41 +87,37 @@ function TransferNFTCard({ objectId }: TransferProps) {
...
 
     const handleOnClearSubmitError = useCallback(() => {
         setSendError(null);
     }, []);
 
-    const TransferNFT = (
-        <>
+    return (
+        <div className={st.container}>
+            {' '}
```
Comment (Jordan-Mysten) [inline wallet/src/ui/app/pages/home/nft-details/transfer-nft/index.tsx:112]: Is this space supposed to be here?

File: wallet/src/ui/app/components/transactions-card/index.tsx [hunk 4]
```diff
@@ -3,64 +3,105 @@
...
+    );
+
+    const transferStatus = txn.status === 'success' ? 'Checkmark' : 'Close';
+    const TxIcon = txn.isSender ? SuiIcons.ArrowLeft : SuiIcons.Buy;
+    const iconClassName = txn.isSender
+        ? cl(st.arrowActionIcon, st.angledArrow)
+        : cl(st.arrowActionIcon, st.buyIcon);
+
+    const date = txn?.timestamp_ms
+        ? new Date(txn.timestamp_ms).toLocaleDateString('en-us', {
```
Comment (Jordan-Mysten) [inline wallet/src/ui/app/components/transactions-card/index.tsx:37]: If you make the date formatting helper not a hook, you could leverage it here.

File: wallet/src/ui/app/hooks/useFormatDate.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO - handle multiple date formats
+// Wed Aug 05
+export default function useFormatDate(date: Date) {
```
Comment (Jordan-Mysten) [inline wallet/src/ui/app/hooks/useFormatDate.ts:6]: This likely doesn't need to be a hook, I can't really imagine anything that'd need to be done here that would tie into react?

File: wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 6]
```diff
@@ -27,6 +31,14 @@ export type TxResultState = {
     txGas: number;
     kind: TransactionKindName | undefined;
     From: string;
+    Amount?: number;
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/txresults/index.ts:34]:
```suggestion
    amount?: number;
```

File: wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 7]
```diff
@@ -27,6 +31,14 @@ export type TxResultState = {
...
+    timestamp_ms?: number;
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/txresults/index.ts:35]:
```suggestion
    timestampInMs?: number;
```

File: wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 8]
```diff
@@ -92,8 +107,19 @@ export const getTransactionsByAddress = createAsyncThunk<
                             }
                             const txn = txns[0];
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/txresults/index.ts:108]: Could you add a TODO comment here to handle batch transaction?

File: wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 9]
```diff
@@ -92,8 +107,19 @@ export const getTransactionsByAddress = createAsyncThunk<
...
                             const txKind = getTransactionKindName(txn);
+
+                            const tranferSui =
+                                txKind === 'TransferSui'
+                                    ? getTransferSuiTransaction(txn)
+                                    : null;
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/txresults/index.ts:114]:
```suggestion
                            const tranferSui = getTransferSuiTransaction(txn);
```

see the function definition: https://github.com/MystenLabs/sui/blob/6173c3a8ba0c3179ce800b448fb841f925195212/sdk/typescript/src/types/transactions.ts#L237

File: wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 10]
```diff
@@ -92,8 +107,19 @@ export const getTransactionsByAddress = createAsyncThunk<
...
                             const recipient =
-                                getTransferObjectTransaction(txn)?.recipient;
+                                txKind === 'TransferSui'
+                                    ? tranferSui?.recipient
+                                    : getTransferObjectTransaction(txn)
+                                          ?.recipient;
+
+                            const txTransferObject =
+                                getTransferObjectTransaction(txn);
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/txresults/index.ts:122]:
```suggestion
                                tranferSui?.recipient ?? txTransferObject?.recipient;
                             // Move this to line 114
                            const txTransferObject =
                                getTransferObjectTransaction(txn);
```

File: wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 11]
```diff
@@ -102,6 +128,19 @@ export const getTransactionsByAddress = createAsyncThunk<
                                 txGas: getTotalGasUsed(txEff),
                                 kind: txKind,
                                 From: res.data.sender,
+                                ...(txTransferObject?.objectRef.objectId
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/txresults/index.ts:131]:
```suggestion
                                ...(txTransferObject?
```

File: wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 12]
```diff
@@ -115,7 +154,39 @@ export const getTransactionsByAddress = createAsyncThunk<
...
+
+        // Get all objectId and batch fetch objects for transactions with objectIds
+        // remove duplicates
+        const objectIDs = [
+            ...new Set(
+                resp.filter((itm) => itm).map((itm) => itm?.objectId || '')
+            ),
+        ];
+
+        const getObjectBatch = await dispatch(batchBetchObjects(objectIDs));
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/txresults/index.ts:166]:
```suggestion
        const getObjectBatch = await dispatch(batchFetchObjects(objectIDs));
```

File: wallet/src/ui/app/redux/slices/sui-objects/index.ts [hunk 13]
```diff
@@ -46,6 +46,22 @@ export const fetchAllOwnedObjects = createAsyncThunk<
     return allSuiObjects;
 });
 
+export const batchBetchObjects = createAsyncThunk<
```
Comment (666lcz) [inline wallet/src/ui/app/redux/slices/sui-objects/index.ts:49]:
```suggestion
export const batchFetchObjects = createAsyncThunk<
```

File: wallet/src/ui/app/hooks/useFormatDate.ts (hunk 1 above)
Comment (Jibz1) [reply wallet/src/ui/app/hooks/useFormatDate.ts:7]: Still working on this. Thanks for the feedback

File: wallet/src/ui/app/redux/slices/sui-objects/index.ts (hunk 13 above)
Comment (Jibz1) [reply wallet/src/ui/app/redux/slices/sui-objects/index.ts]: This is done

File: wallet/src/ui/app/pages/home/receipt/index.tsx (hunk 2 above)
Comment (Jibz1) [reply wallet/src/ui/app/pages/home/receipt/index.tsx]: Thanks done.

File: wallet/src/ui/app/components/transactions-card/index.tsx (hunk 4 above)
Comment (Jibz1) [reply wallet/src/ui/app/components/transactions-card/index.tsx]: Done.

File: wallet/src/ui/app/hooks/useFormatDate.ts (hunk 1 above)
Comment (Jibz1) [reply wallet/src/ui/app/hooks/useFormatDate.ts]: This is done, review

File: wallet/src/ui/app/components/transactions-card/index.tsx [hunk 14]
```diff
@@ -3,64 +3,109 @@
...
+        ? cl(st.arrowActionIcon, st.angledArrow)
+        : cl(st.arrowActionIcon, st.buyIcon);
+
+    const date = txn?.timestampMs
+        ? formatDate(txn.timestampMs, [
+              'weekday',
+              'month',
+              'day',
+              // 'hour',
+              // 'minute',
```
Comment (666lcz) [inline wallet/src/ui/app/components/transactions-card/index.tsx:45]: do we need this?

**Review by 666lcz — APPROVED** [review]

File: wallet/src/ui/app/components/transactions-card/index.tsx (hunk 14 above)
Comment (Jibz1) [reply wallet/src/ui/app/components/transactions-card/index.tsx:45]: Based on the design we do,

**Pull request merged**